src/core/absorption_chiller.jl

```diff
@@ -28,61 +28,217 @@
+    for (k, v) in d
+        if typeof(v) <: AbstractVector{Any}
+            d[k] = convert(Vector{Float64}, v)  # JSON.parsefile makes things Vector{Any}
+        end
+    end
+
+    absorp_chl = AbsorptionChiller(; dictkeys_tosymbols(d)...)
+
+    #check for 0.0 max size, return nothing if so
+    if absorp_chl.max_ton == 0.0
+        @warn "0.0 kW provided as capacity for AbsoprtionChiller, this technology will be excluded."
+        return nothing
+    end
+
+    custom_ac_inputs = Dict{Symbol, Any}(
+        :hot_water_or_steam => absorp_chl.hot_water_or_steam,
+        :installed_cost_per_ton => absorp_chl.installed_cost_per_ton,
+        :cop_thermal => absorp_chl.cop_thermal,
+        :om_cost_per_ton => absorp_chl.om_cost_per_ton
+    )
+
+    htf_defaults_response = get_absorption_chiller_defaults(;max_ton=absorp_chl.max_ton,
+        hot_water_or_steam=absorp_chl.hot_water_or_steam, 
+        chp_prime_mover=chp_prime_mover, 
+        existing_boiler=existing_boiler
+    )
+    
+    #convert defaults for any properties not enetered
+    defaults = htf_defaults_response["default_inputs"]
+    for (k, v) in custom_ac_inputs
+        if isnothing(v)
+            setproperty!(absorp_chl, k, defaults[string(k)])
+        end
+    end
+
+    # generate derived inputs for use in JuMP model
+    absorp_chl.min_kw = absorp_chl.min_ton * KWH_THERMAL_PER_TONHOUR
+    absorp_chl.max_kw = absorp_chl.max_ton * KWH_THERMAL_PER_TONHOUR
+    absorp_chl.installed_cost_per_kw = absorp_chl.installed_cost_per_ton / KWH_THERMAL_PER_TONHOUR
+    absorp_chl.om_cost_per_kw = absorp_chl.om_cost_per_ton / KWH_THERMAL_PER_TONHOUR
+
+    return absorp_chl
+end
+
+
+"""
+get_absorption_chiller_defaults(prime_mover::String, boiler_type::String, size_class::Int)
 
-        min_kw = min_ton * KWH_THERMAL_PER_TONHOUR
-        max_kw = max_ton * KWH_THERMAL_PER_TONHOUR
-        installed_cost_per_kw = installed_cost_per_ton / KWH_THERMAL_PER_TONHOUR
-        om_cost_per_kw = om_cost_per_ton / KWH_THERMAL_PER_TONHOUR
-
-        new(
-            min_ton,
-            max_ton,
-            cop_thermal,
-            cop_electric,
-            installed_cost_per_ton,
-            om_cost_per_ton,
-            macrs_option_years,
-            macrs_bonus_fraction,
-            min_kw,
-            max_kw,
-            installed_cost_per_kw,
-            om_cost_per_kw
+return a Dict{String, Any} by selecting the appropriate values from 
+data/chp/absorption_chiller_defaults.json, which contains values based on heat transfer medium (hot_water_or_steam)
+such as:
+- "installed_cost_per_ton"
+- "om_cost_per_ton"
+- "cop_thermal"
+- "tech_sizes_for_cost_curve"
+
+Unlike CHP, the AbsorptionChiller tech sizes inform a single rate for installed_cost_per_ton that uses max_ton as input;
+there is no piecewise linear cost curve for the AbsorptionChiller technology.
+
+Inputs: 
+max_ton::Float64 -- maximum size of the absorption chiller technology, in tons
```

Comment:
The defaults for AbsorptionChiller should actually be based on the max cooling load in tons, not the input value of max_ton for AbsorptionChiller. So we'll have to feed that in from the CoolingLoad. I mention in another comment that I think the max_ton value here should actually default to a big number like the other techs.

Reply:
Thanks for catching this!  The bulk of this is covered in fe94dee33392ed199b4a97dd1c5009f5dfefbfb4 and other adjacent updates are in cb45aef5411d96cc4d3a97849319176180972e81 and 5faeaed52d3a7e9532277aa96481b34b90f48768.  I updated the relevant tests as well in a966e817b2c61ad25eb620a3ff122b7b93890fdf but will wait on them passing before I send back to you for another review.